src/core/src/core_logic/PatchAssessor.py

```diff
@@ -27,14 +27,15 @@ def __init__(self, env_layer, execution_config, composite_logger, telemetry_writ
 
         self.composite_logger = composite_logger
         self.telemetry_writer = telemetry_writer
+        self.is_telemetry_setup = False
```

Comment:
Why is this necessary? All calls to write telemetry are blind to whether telemetry is turned on.

Unclear about line 38 too. Will leave a comment there.